pkg/sys/mount/listener_linux.go

```diff
@@ -0,0 +1,256 @@
+	listener.mutex.Lock()
+	defer listener.mutex.Unlock()
+
+	if len(listener.eventMap) == 0 && len(listener.keys) > 0 {
+		listener.keys = []string{}
+	}
+
+	var event *Event
+	var found bool
+	for !found {
+		if len(listener.keys) == 0 {
+			break
+		}
+
+		key := listener.keys[0]
+		listener.keys = listener.keys[1:]
+		if event, found = listener.eventMap[key]; found {
+			delete(listener.eventMap, key)
+			return key, event, nil, nil
+		}
+	}
+
+	// As no event found, wait for event to be set.
+	waitCh := listener.waitCh
+	if waitCh == nil {
+		waitCh = make(chan struct{})
+		listener.waitCh = waitCh
+	}
+	return "", nil, waitCh, nil
+}
+
+func (listener *Listener) Get(ctx context.Context) (string, *Event, error) {
+	for {
+		key, event, waitCh, err := listener.get()
+		switch {
+		case err != nil:
+			return "", nil, err
+		case event != nil && key != "":
+			listener.cachedMounts[key] = event.latest
+			if len(event.latest) == 0 {
+				// no mounts found for this maj:min in /proc/1/mountinfo (umounted)
+				delete(listener.cachedMounts, key)
+			}
+			return key, event, nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return "", nil, fmt.Errorf("cancelled by context; %w", ctx.Err())
+		case <-listener.closeCh:
+			return "", nil, errors.New("closed listener")
+		case <-waitCh:
+		}
+	}
+}
+
+func (listener *Listener) set(majMin string, event *Event) {
+	listener.mutex.Lock()
+	defer listener.mutex.Unlock()
+
+	listener.keys = append(listener.keys, majMin)
+	listener.eventMap[majMin] = event
+
+	if listener.waitCh != nil {
+		close(listener.waitCh)
+		listener.waitCh = nil
+	}
+}
+
+func (listener *Listener) start() {
+	go func() {
+		events := make([]syscall.EpollEvent, 16)
+		defer listener.Close()
+		for {
+			select {
+			case <-listener.closeCh:
+				return
+			default:
+				if err := syscall.SetNonblock(int(listener.mountFile.Fd()), true); err != nil {
+					klog.Infof("Error while waiting for mount events: %v", err)
+					continue
+				}
+				_, err := syscall.EpollWait(listener.epollFd, events[:], int(10*time.Minute/time.Millisecond))
+				if err != nil {
+					klog.Infof("EPOLLWAIT ERR: %v", err)
+					continue
+				}
+				probedMounts, err := ProbeMounts()
+				if err != nil {
+					klog.Infof("Error while probing mounts while monitoring mount events: %v", err)
+					continue
+				}
+				if diff := getMountDiffs(probedMounts, listener.cachedMounts); diff.MountsChanged() {
+					eventMap := toEventMap(diff, probedMounts)
+					for k, v := range eventMap {
+						listener.set(k, v)
+					}
+				}
+			}
+		}
+	}()
+}
+
+func StartListener() (*Listener, error) {
+	epfd, err := syscall.EpollCreate1(0)
+	if err != nil {
+		return nil, err
+	}
+	f, err := os.Open(mountInfoProcFile)
+	if err != nil {
+		return nil, err
+	}
+
+	err = syscall.EpollCtl(epfd, syscall.EPOLL_CTL_ADD, int(f.Fd()), &syscall.EpollEvent{
+		Events: syscall.EPOLLIN | (syscall.EPOLLET & 0xffffffff),
```

Comment:
`syscall.EPOLLET & 0xffffffff` returns `syscall.EPOLLET`. What are you trying to do here?